tests/test_template_alias_base.cpp

```diff
@@ -0,0 +1,24 @@
+#include "pybind11_tests.h"
+
+#include <memory>
+
+template <class T>
+using DefaultAllocator = std::allocator<T>;
+
+template <template <class> class Allocator = DefaultAllocator>
+struct Base {};
+
+template <template <class> class Allocator = DefaultAllocator>
+struct S : public Base<Allocator> {};
+
+// this returns S<DefaultAllocator> even though we register
+// the type S<std::allocator>
+// MSVC and Clang on Windows failed here to detect the registered type
+S<> make_S() { return S<>{}; }
+
+TEST_SUBMODULE(template_alias_base, m) {
+    py::class_<Base<std::allocator>>(m, "B_std");
+    py::class_<S<std::allocator>, Base<std::allocator>>(m, "S_std");
```

Comment:
Hm, my example does not trigger the issue I see.

In my downstream project, `Base` is implemented in a separate module...